R/human_infection.R

```diff
@@ -289,29 +295,61 @@ calculate_treated <- function(
       replace = TRUE
     )
   ])
-
-  successful <- bernoulli_multi_p(parameters$drug_efficacy[drugs])
-  treated_index <- bitset_at(seek_treatment, successful)
-
-  # Update those who have been treated
-  if (treated_index$size() > 0) {
-    variables$state$queue_update('Tr', treated_index)
+  
+  if(parameters$antimalarial_resistance == TRUE) {
+    antimalarial_resistance_drug_index <- as.numeric(parameters$antimalarial_resistance_drug)[drugs]
+    artemisinin_resistance_proportion <- vector()
+    early_treatment_failure_probability <- vector()
+    drug_indices <- list(); drug_index <- vector()
+    
+    for(i in seq_along(parameters$antimalarial_resistance_drug)) {
+      drug_indices[[i]] <- which(drugs == i)
+      drug_index[i] <- which(parameters$antimalarial_resistance_drug == i)
+      matched_t <- match_timestep(ts = parameters$antimalarial_resistance_timesteps[[drug_index[i]]], t = timestep)
+      artemisinin_resistance_proportion[drug_indices[[i]]] <- parameters$prop_artemisinin_resistant[[drug_index[i]]][matched_t]
+      early_treatment_failure_probability[drug_indices[[i]]] <- parameters$early_treatment_failure_prob[[drug_index[i]]][matched_t]
+    }
```

Comment:
Great spot! 

I have replaced the for-loop that retrieves the resistance parameters for each drug assigned to individuals receiving treatment with a function called `get_antimalarial_resistance_parameters()` (antimalarial_resistance.R). The function takes as it's inputs the `parameters` list, the vector of `drugs`, and the current timestep, and returns a list of vectors, each of length `length(drugs)`, of each of the antimalarial resistance parameters corresponding to the drugs administered and the current timestep. To avoid issues arising when users do not assign resistance to all drugs for which clinical treatment is specified, within `get_antimalarial_resistance_parameters()` the vectors are initiated with no-resistance defaults (0's for the resistance proportions / probabilities, `parameters$dt` for the slow parasite clearance dt). The function then loops through each of the drugs for which resistance has been parameterised (`parameters$antimalarial_resistance_drugs`), gets the indices from `drugs` which matches the drug, and uses these to assign the appropriate resistance parameters to the output vectors.

To support the use of `get_antimalarial_resistance_parameters()`, which was pulled from the antimalarial_resistance/spc branch, I have also updated the `set_antimalarial_resistance()` function to include `slow_parasite_clearance_time` as an input.

 I have added some additional tests for the `get_antimalarial_resistance_parameters()` function in test-antimalarial-resistance.R and updated those that existed for `set_antimalarial_resistance()` so that they are compatible with the updated version of the function. I have also gone through the unit tests I'd written in test-infection-integration.R to make them compatible with these changes. I have also added the additional test recommended above, in which resistance is only set to one of multiple drugs to test-infection-integration.R. Finally, I have updated the documentation in parameters.R for `get_parameters()` to include an entry for `dt_slow_parasite_clearance` and have updated the vignette (Antimalarial_Resistance.Rmd) to explain the new `dt_slow_parasite_clearance` parameter and ensure it's compatible with the new version of `set_antimalarial_resistance()`